modules/objdetect/src/qrcode.cpp

```diff
@@ -2369,7 +2369,8 @@ bool QRDecode::samplingForVersion()
     CV_TRACE_FUNCTION();
     const double multiplyingFactor = (version < 3)  ? 1 :
                                      (version == 3) ? 1.5 :
-                                     version * (version + 1);
+                                     (version < 6) ? version*(version+1) :
+                                     version;
```

Comment:
> `version;`

`version * 5` ?

Reply:
fixing the tail processing process allows to reduce `multiplyingFactor` to 1 and skip strange`resize()` step

I want to add a few more tests and skip this step or reduce the `multiplyingFactor` to 2, for example.

Reply:
now `multiplyingFactor`:
```
    const double multiplyingFactor = (version < 3)  ? 1. :
                                     (version == 3) ? 2. :
                                     3.;
```